### PR TITLE
Set engine to be greater than node version 16

### PR DESCRIPTION
**Pull request by supl**
This PR sets the `engine` field to `>=16.0.0` in the package.json file.
It means that we would like to ask the developers to use the latest LTS.

It also upgrades the lockfileVersion to `2` for a faster installation.

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
@supl The CI fails. Can you check?

**Comment by supl** [comment]
> @supl The CI fails. Can you check?

Somehow CircleCI couldn't finish the test.
It seems that some container(s) was not up when auditor-related test were running after the Scalar DL network has been restarted.

I will by this chance switch the CI to GitHub Actions.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          echo -n "#"
+        done
+        echo ""
+        docker cp function_schema.cql scalardl-samples-cassandra-1:function_schema.cql
+        docker exec scalardl-samples-cassandra-1 cqlsh -f function_schema.cql
+        docker-compose -f docker-compose.yml -f expose-cassandra-port.yml up scalardl-ledger-schema-loader-cassandra
+        docker-compose -f docker-compose.yml -f expose-cassandra-port.yml up -d scalar-ledger ledger-envoy
+        sleep 5
+
+    - name: NPM Install
```
Comment (ypeckstadt) [inline .github/workflows/e2e.yml:50]: Maybe can the node modules package cache feature to speed up the `npm install` process?

```
- name: Cache dependencies
        uses: actions/cache@v2
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-
```

**Review by ypeckstadt — APPROVED** [review]
LGTM. Just left one small comment/suggestion.

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (supl) [reply .github/workflows/e2e.yml:53]: Thank you, Yves.
I found that maybe I can leverage the `setup-node` action's built-in caching feature.
https://github.com/actions/setup-node

I will do that.

**Comment by supl** [comment]
@feeblefakie 
I migrated the CI workflow from CirCleCI to GitHub Actions.
Don't worry about the failure of the CircleCI workflow.
That's because I deleted the `.circleci` folder.

Could you:
- Set up a secret named `NPM_TOKEN` with your NPM registry token? (it will be used when we add new tag)
- Click `Stop Building` button on https://app.circleci.com/settings/project/github/scalar-labs/scalardl-node-client-sdk?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fscalar-labs%2Fscalardl-node-client-sdk or we can just delete the project.

Thank you :)

**Review by feeblefakie — COMMENTED** [review]
The PR seems to have more changes than what the title says, but is this what you mean?
Adding e2e and publishing workflows has something to do with the engine version?

**Comment by feeblefakie** [comment]
@supl Also, the CI is failing so can you check?

**Comment by supl** [comment]
@feeblefakie 
I am so sorry, I must have made you get lost to follow the comments.

In the beginning, this PR was only for setting the Node.js engine.

However, we couldn't pass the CircleCI test.

It has been an old issue of hardware resources.
I guess that is because we have two end-to-end tests and we restart the local Scalar DL network in one CircleCI job.

Previously, we always rerun the test until it can finish the workflow.
But, somehow in this time, no matter how many times I rerun the test, it always fails.

So, I think maybe by this chance I also fix the issue and migrate the CI to GitHub Actions.

Sorry, it's too confusing.
If it's ok for you, I will create another PR and move the migration part from this PR to there.
After the migration PR is merged, then we can come back to this PR.

**Pull request merged**